### DISCUSSION TITLE
How to import raw sigma yml file to Playbook

### DISCUSSION THREAD

**Discussion by nguyenl95** [Security-Onion-Solutions/securityonion#12585] [2.4]
### Version

2.4.40

### Installation Method

Security Onion ISO image

### Description

configuration

### Installation Type

Standalone

### Location

cloud

### Hardware Specs

Meets minimum requirements

### CPU

4

### RAM

16

### Storage for /

200

### Storage for /nsm

50

### Network Traffic Collection

other (please provide detail below)

### Network Traffic Speeds

Less than 1Gbps

### Status

Yes, all services on all nodes are running OK

### Salt Status

No, there are no failures

### Logs

No, there are no additional clues

### Detail

How to import raw sigma yml file to Playbook

Hello, I have several custom sigma files (for example: https://github.com/joesecurity/sigma-rules).
How can I import them?

### Guidelines

- [X] I have read the discussion guidelines at https://github.com/Security-Onion-Solutions/securityonion/discussions/1720 and assert that I have followed the guidelines.

**Comment by nguyenl95** [comment]
@cm-ops how i can import those yaml files automatically

**Reply by cm-ops** [reply]
No need to call out. You could try and put the `.yml` files in` /opt/so/conf/soctopus/sigma-import` then run `so-playbook-import`.

A couple of caveats: one it is not fully supported to do it this way as it will not dedup plays and two if you have existing IDH yamls in there, remove them so they are not imported again. 

If you only want a few plays, I would suggest you import them via `Create New Play` in the web interface.

**Reply by nguyenl95** [reply]
@cm-ops thanks for your quick response.
Is it also possible/correct way to have file [playbook_bulk_update.py](https://github.com/Security-Onion-Solutions/securityonion-image/blob/e80bca63bfdce58d24421a422d6bf88461698bcf/so-soctopus/so-soctopus/playbook_bulk-update.py#L107) modified to apply new `.yml` files in custom folders (also applying rule dedup)?

**Reply by cm-ops** [reply]
Currently the only supported way of adding custom sigma rules is via the web console and `Create New Play`. 

Having said that, this is something that is currently being looked at.
